homeassistant/components/conversation/default_agent.py

```diff
@@ -181,9 +183,7 @@ async def async_recognize(
         lang_intents = self._lang_intents.get(language)
 
         # Reload intents if missing or new components
-        if lang_intents is None or (
-            lang_intents.loaded_components - self.hass.config.components
```

Comment:
Don’t we still need this for custom components

Reply:
Custom components can supply intents, but not sentences so we should still be good. If you have a sentence that uses an intent from a custom component, the sentence will get loaded regardless of whether the custom component is loaded or not.